### PR TITLE
A script that generates 3 images

### PR DESCRIPTION
**Pull request by akoshelev**
This script also performs key exchange between those images and updates them in-place before generating tarballs that are ready to be distributed. 

Tested it on an internal AWS cluster

### Helper setup

```bash
# h1
sudo docker run -d -it -p 443:443 private-attribution/ipa:c8169e4805-h1 /usr/local/bin/ipa-helper --network /etc/ipa/pub/network.toml --identity 1 --tls-cert /etc/ipa/pub/h1.pem --tls-key /etc/ipa/h1.key --port 443

# h2
sudo docker run -d -it -p 443:443 private-attribution/ipa:c8169e4805-h2 /usr/local/bin/ipa-helper --network /etc/ipa/pub/network.toml --identity 2 --tls-cert /etc/ipa/pub/h2.pem --tls-key /etc/ipa/h2.key --port 443

# h3
sudo docker run -d -it -p 443:443 private-attribution/ipa:c8169e4805-h3 /usr/local/bin/ipa-helper --network /etc/ipa/pub/network.toml --identity 3 --tls-cert /etc/ipa/pub/h3.pem --tls-key /etc/ipa/h3.key --port 443
```

### Client setup

Almost automated, except `network.toml` generated had to be delivered manually. 

```
cargo run --bin test_mpc --features="cli test-fixture" -- gen-ipa-inputs -n 1000000 --output-file /tmp/ipa-events.txt --max-breakdown-key 20

cargo run --bin test_mpc --features="cli test-fixture" -- --network network.toml --input-file /tmp/ipa-events.txt --field fp32-bit-prime semi-honest-ipa --max-breakdown-key 20

 INFO test_mpc:
+-----+-------------+-------------+-------+
| Row | Expected    | Actual      | Diff? |
+=========================================+
| 0   | Some(25727) | Some(25727) |       |
|-----+-------------+-------------+-------|
| 1   | Some(26212) | Some(26212) |       |
|-----+-------------+-------------+-------|
| 2   | Some(26642) | Some(26642) |       |
|-----+-------------+-------------+-------|
| 3   | Some(26237) | Some(26237) |       |
|-----+-------------+-------------+-------|
| 4   | Some(26814) | Some(26814) |       |
|-----+-------------+-------------+-------|
| 5   | Some(26455) | Some(26455) |       |
|-----+-------------+-------------+-------|
| 6   | Some(26714) | Some(26714) |       |
|-----+-------------+-------------+-------|
| 7   | Some(25851) | Some(25851) |       |
|-----+-------------+-------------+-------|
| 8   | Some(26358) | Some(26358) |       |
|-----+-------------+-------------+-------|
| 9   | Some(26142) | Some(26142) |       |
|-----+-------------+-------------+-------|
| 10  | Some(26324) | Some(26324) |       |
|-----+-------------+-------------+-------|
| 11  | Some(26266) | Some(26266) |       |
|-----+-------------+-------------+-------|
| 12  | Some(26332) | Some(26332) |       |
|-----+-------------+-------------+-------|
| 13  | Some(26056) | Some(26056) |       |
|-----+-------------+-------------+-------|
| 14  | Some(27001) | Some(27001) |       |
|-----+-------------+-------------+-------|
| 15  | Some(26078) | Some(26078) |       |
|-----+-------------+-------------+-------|
| 16  | Some(26246) | Some(26246) |       |
|-----+-------------+-------------+-------|
| 17  | Some(26288) | Some(26288) |       |
|-----+-------------+-------------+-------|
| 18  | Some(27035) | Some(27035) |       |
|-----+-------------+-------------+-------|
| 19  | Some(26346) | Some(26346) |       |
+-----+-------------+-------------+-------+

```

### CODE REVIEW

File: scripts/helper-image.sh [hunk 1]
```diff
@@ -3,11 +3,14 @@
...
 
 help() {
-  echo "Usage: $0 --hostname [hostname] --identity [1|2|3]"
+  echo "Usage: $0 --hostname [hostname] --identity [1|2|3] --platform [platform]"
   echo "- hostname: public hostname that will appear on TLS certificate"
   echo "- identity: helper identity to be fixed inside the IPA build"
+  echo "- platform: the platform for the helper image, defaults to linux/amd64"
 }
 
+platform="linux/amd64"
```
Comment (akoshelev) [inline scripts/helper-image.sh:12]: that was a gotcha for me - docker uses supervisor's platform when building images, so I couldn't run them on Linux later. 

Avoid generating those images on MacOS - it is extremely slow (but working). Better to run this script on a Linux machine

Comment (richajaindce) [inline scripts/helper-image.sh:12]: We dont need this line right?

Comment (akoshelev) [reply scripts/helper-image.sh:12]: mm, can you elaborate more? I am not sure I understand the source for confusion here

File: docker/helper.Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # syntax=docker/dockerfile:1
 ARG SOURCES_DIR=/usr/src/ipa
 FROM rust:latest as builder
-
+ARG SOURCES_DIR
```
Comment (akoshelev) [inline docker/helper.Dockerfile:4]: Not surprised, but Linux version has slightly different expectations (not sure why though)

**Comment by richajaindce** [comment]
Lets go ahead and merge this!

**Review by richajaindce — APPROVED** [review]

**Pull request merged**